--- a/lib/raop.c
+++ b/lib/raop.c
@@ -849 +849 @@
-    httpd_stop(raop->httpd);
+    if (raop->httpd) httpd_stop(raop->httpd);
--- a/lib/dnssd.c
+++ b/lib/dnssd.c
@@ -281,0 +282,2 @@
+        free(dnssd->name);
+        free(dnssd->hw_addr);
@@ -455,4 +456,0 @@
-    if (dnssd->airplay_service == NULL) {
-        free(dnssd->name);
-        free(dnssd->hw_addr);
-    }
@@ -476,4 +473,0 @@
-    if (dnssd->raop_service == NULL) {
-        free(dnssd->name);
-        free(dnssd->hw_addr);
-    }
